crypto/evp/p_lib.c

```diff
@@ -1963,6 +1963,7 @@ void *evp_pkey_export_to_provider(EVP_PKEY *pk, OSSL_LIB_CTX *libctx,
             goto end;
         }
         EVP_KEYMGMT_free(tmp_keymgmt); /* refcnt-- */
+        tmp_keymgmt = NULL;
```

Comment:
This is not correct. The original code was right. Note that the an EVP_KEYMGMT is ref counted. `EVP_KEYMGMT_free` only actually frees if the ref count drops to 0. However on line 1949 we explicitly up ref the object (so we expect there to be at least 2 refs now). This free is just the corresponding down ref, so the ref count should now be at least 1, i.e. the `tmp_keymgmt` object has not really been freed and is still valid for use.